### PR TITLE
Fix  bugs found in testing multi spring boot version

### PR DESCRIPTION
**Pull request by moarychan**
Found some bugs and code smells in testing this PR https://github.com/Azure/azure-sdk-for-java/pull/21258.

### CODE REVIEW

File: sdk/boms/azure-spring-cloud-dependencies/pom.xml [hunk 1]
```diff
@@ -153,11 +153,6 @@
         <artifactId>azure-spring-cloud-starter-servicebus</artifactId>
         <version>${azure.spring.cloud.version}</version>
       </dependency>
-      <dependency>
-        <groupId>com.azure.spring</groupId>
-        <artifactId>azure-spring-cloud-storage</artifactId>
-        <version>${azure.spring.cloud.version}</version>
-      </dependency>
```
Comment (saragluna) [inline sdk/boms/azure-spring-cloud-dependencies/pom.xml:160]: Why is this getting removed?

File: sdk/spring/spring-test-template.yml [hunk 2]
```diff
@@ -10,6 +10,8 @@ parameters:
     - spring/azure-spring-cloud-test-eventhubs
     - spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder
     - spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka
+    - spring/azure-spring-boot-test-aad
+    - spring/azure-spring-boot-test-aad-b2c
```
Comment (saragluna) [inline sdk/spring/spring-test-template.yml:14]: I think there are no testing resources defined in aad and aad b2c project?

File: sdk/spring/tests.yml [hunk 3]
```diff
@@ -9,6 +9,7 @@ stages:
         Clouds: China
         TestResourceDirectories:
           - spring/azure-spring-boot-test-aad
+          - spring/azure-spring-boot-test-aad-b2c
```
Comment (saragluna) [inline sdk/spring/tests.yml:12]: do we need this?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/boms/azure-spring-cloud-dependencies/pom.xml (hunk 1 above)
Comment (moarychan) [reply sdk/boms/azure-spring-cloud-dependencies/pom.xml]: I found the duplicate dependency.

File: sdk/spring/spring-test-template.yml (hunk 2 above)
Comment (moarychan) [reply sdk/spring/spring-test-template.yml:14]: oh, I understand wrong, so we can remove test-aad too.

File: sdk/spring/tests.yml (hunk 3 above)
Comment (moarychan) [reply sdk/spring/tests.yml:12]: let me remove it.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**